Index: src/drivers/net/ethernet/qlogic/qed/Makefile
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/Makefile	2017-04-19 17:52:09.637632101 +0200
+++ src/drivers/net/ethernet/qlogic/qed/Makefile	2017-04-19 18:30:08.599148664 +0200
@@ -1,3 +1,10 @@
+CONFIG_QED := m
+CONFIG_QED_SRIOV := y
+CONFIG_QED_LL2 := y
+CONFIG_QED_RDMA := y
+CONFIG_QED_ISCSI := y
+CONFIG_QED_FCOE := y
+
 obj-$(CONFIG_QED) := qed.o
 
 qed-y := qed_cxt.o qed_dev.o qed_hw.o qed_init_fw_funcs.o qed_init_ops.o \
