Index: src/drivers/net/ethernet/qlogic/qed/qed_main.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_main.c	2017-04-20 18:29:53.402039442 +0200
+++ src/drivers/net/ethernet/qlogic/qed/qed_main.c	2017-04-20 18:30:03.061924286 +0200
@@ -73,7 +73,8 @@
 	__stringify(FW_MAJOR_VERSION) "."	\
 	__stringify(FW_MINOR_VERSION) "."	\
 	__stringify(FW_REVISION_VERSION) "."	\
-	__stringify(FW_ENGINEERING_VERSION)
+	__stringify(FW_ENGINEERING_VERSION)	\
+	FW_RELEASE_SUFFIX
 
 #define QED_FW_FILE_NAME	\
 	"qed/qed_init_values_zipped-" FW_FILE_VERSION ".bin"
Index: src/include/linux/qed/common_hsi.h
===================================================================
--- src.orig/include/linux/qed/common_hsi.h	2017-04-20 18:30:03.046924465 +0200
+++ src/include/linux/qed/common_hsi.h	2017-04-20 18:30:03.062924274 +0200
@@ -105,6 +105,7 @@
 #define FW_MINOR_VERSION	15
 #define FW_REVISION_VERSION	3
 #define FW_ENGINEERING_VERSION	0
+#define FW_RELEASE_SUFFIX	"_dup7.3"
 
 /***********************/
 /* COMMON HW CONSTANTS */
