Index: src/drivers/net/ethernet/qlogic/qed/qed_l2.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_l2.c	2017-04-26 20:47:46.666611203 +0200
+++ src/drivers/net/ethernet/qlogic/qed/qed_l2.c	2017-04-26 20:48:15.610269614 +0200
@@ -2391,10 +2391,20 @@
 {
 	return &qed_eth_ops_pass;
 }
-EXPORT_SYMBOL(qed_get_eth_ops);
+
+const struct qed_eth_ops *qed_ql41000_get_eth_ops(void)
+{
+	return qed_get_eth_ops();
+}
+EXPORT_SYMBOL(qed_ql41000_get_eth_ops);
 
 void qed_put_eth_ops(void)
 {
 	/* TODO - reference count for module? */
 }
-EXPORT_SYMBOL(qed_put_eth_ops);
+
+void qed_ql41000_put_eth_ops(void)
+{
+	qed_put_eth_ops();
+}
+EXPORT_SYMBOL(qed_ql41000_put_eth_ops);
