Index: src/drivers/net/ethernet/qlogic/qed/qed_mcp.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_mcp.c	2017-12-01 20:46:11.841915550 +0100
+++ src/drivers/net/ethernet/qlogic/qed/qed_mcp.c	2017-12-01 20:46:48.341410818 +0100
@@ -1092,11 +1092,20 @@
 	case FW_MB_PARAM_GET_PF_RDMA_ROCE:
 		*p_proto = QED_PCI_ETH_ROCE;
 		break;
-	case FW_MB_PARAM_GET_PF_RDMA_BOTH:
-		DP_NOTICE(p_hwfn,
-			  "Current day drivers don't support RoCE & iWARP. Default to RoCE-only\n");
+	case FW_MB_PARAM_GET_PF_RDMA_BOTH: {
+		static int has_printed;
+
+		if (!has_printed) {
+			DP_NOTICE(p_hwfn,
+				  "Current day drivers don't support "
+				  "RoCE & iWARP. Default to RoCE-only\n");
+
+			has_printed = 1;
+		}
+
 		*p_proto = QED_PCI_ETH_ROCE;
 		break;
+	}
 	case FW_MB_PARAM_GET_PF_RDMA_IWARP:
 	default:
 		DP_NOTICE(p_hwfn,
